app/code/Magento/InventoryInStorePickup/Model/Source/InitPickupLocationExtensionAttributes.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickup\Model\Source;
+
+use Magento\Framework\Api\ExtensionAttributesFactory;
+use Magento\Framework\DataObject;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\InventoryInStorePickupApi\Api\Data\PickupLocationInterface;
+
+/**
+ * Set store-pickup related source extension attributes
+ */
+class InitPickupLocationExtensionAttributes
+{
+    /**
+     * @var ExtensionAttributesFactory
+     */
+    private $extensionAttributesFactory;
+
+    /**
+     * @param ExtensionAttributesFactory $extensionAttributesFactory
+     */
+    public function __construct(ExtensionAttributesFactory $extensionAttributesFactory)
+    {
+        $this->extensionAttributesFactory = $extensionAttributesFactory;
+    }
+    /**
+     * @param SourceInterface $source
+     */
+    public function execute(SourceInterface $source): void
+    {
+        if (!$source instanceof DataObject) {
+            return;
+        }
+        $pickupAvailable = $source->getData(PickupLocationInterface::IS_PICKUP_LOCATION_ACTIVE);
```

Comment:
In the last review I wasn't clear enough.

Just wanted to make sure that this is the best idea we have to load newly introduced fields.

Reply:
Well, we could do some magic with converting snake_case field name to camelCase, check if the appropriate method exists and then call it...

But I don`t like this idea :)